scrapy/downloadermiddlewares/httpcompression.py

```diff
@@ -83,6 +83,38 @@ def process_response(self, request, response, spider):
 
         return response
 
+    def _handle_encoding(self, body, content_encoding):
+        content_to_decode = body
+
+        separator = b","
+        for encodings in reversed(content_encoding):
+            if separator in encodings:
```

Comment:
What about handling the comma issue earlier? i.e. (untested)

```
encodings = reversed(item.strip() for value in content_encoding for item in value.split(b","))
```

That should simplify the later code.

Reply:
Oh, I see, there’s the editing of `Content-Encoding` to mind as well.